marlowe-playground-client/default.nix

```diff
@@ -32,16 +32,24 @@ let
     packageLockJson = ./package-lock.json;
     githubSourceHashMap = {
       shmish111.nearley-webpack-loader."939360f9d1bafa9019b6ff8739495c6c9101c4a1" = "1brx669dgsryakf7my00m25xdv7a02snbwzhzgc9ylmys4p8c10x";
+      # Note: for some unknown reason, it is necessary to add libxmljs to package.json even though it is a transitive dependency
+      # failure to do this meant that node_modules was being built without it there, even though it is in package-lock.json
       znerol.libxmljs."0517e063347ea2532c9fdf38dc47878c628bf0ae" = "0g3kgwnqfr6v2xp1i7dfbm4z45inz1019ln06lfxl9mwxlc31wfg";
     };
   };
 
   client = buildPursPackage {
-    inherit webCommon nodeModules;
+    inherit pkgs nodeModules;
     src = ./.;
-    checkPhase = "npm run test";
+    checkPhase = ''
+      ${pkgs.nodejs}/bin/npm run test
+    '';
     name = "marlowe-playground-client";
-    psSrc = generated-purescript;
+    extraSrcs = {
+      web-common = webCommon;
+      web-common-marlowe = webCommonMarlowe;
+      generated = generated-purescript;
+    };
```

Comment:
For these `extraSrcs` stanzas, would it make life easier overall if you made the names line up and just did: 

``` nix
  extraSrcs = { inherit webCommon webCommonMarlowe generated-purescript; };
```

(Snake or kebab case, whichever makes most sense. You get the idea...)

Reply:
ah ha, you showed up the slight gotcha here: we use the key as a directory name and so we want to avoid any case-related issues. I put it like this to try (not very hard) to make it a bit more explicit, they key is the directory name, the value is the contents.